### PR TITLE
Amazon-AWS-Modify-Instance-Attribute2

### PR DESCRIPTION
**Pull request by mihaitusa**
Reimplement UpdateInstanceType operation into ModifyInstanceAttribute operation using QueryAPI pattern - part.1.
Signed off by Tusa Mihai: <tusa.mihai@gmail.com>

### CODE REVIEW

**Comment by mihaitusa** [comment]
@Tirla-Alin Please review it!

**Comment by mihaitusa** [comment]
@gituser173 Is ready, tested it, you can do the review when you have time for it....

File: cs-amazon/src/main/java/io/cloudslang/content/amazon/factory/helpers/InstanceUtils.java [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
+import io.cloudslang.content.amazon.entities.aws.InstanceType;
 import io.cloudslang.content.amazon.entities.aws.VolumeType;
 import io.cloudslang.content.amazon.entities.constants.Inputs;
 import io.cloudslang.content.amazon.entities.inputs.InputsWrapper;
 import io.cloudslang.content.amazon.utils.InputsUtil;
 
-import java.util.HashMap;
-import java.util.LinkedHashMap;
-import java.util.Map;
+import java.util.*;
```
Comment (gituser173) [inline cs-amazon/src/main/java/io/cloudslang/content/amazon/factory/helpers/InstanceUtils.java]: import only the ones you use here

File: cs-amazon/src/main/java/io/cloudslang/content/amazon/factory/helpers/InstanceUtils.java [hunk 2]
```diff
@@ -59,24 +65,121 @@
...
+
+        InputsUtil.validateAgainstDifferentArraysLength(deviceNamesArray, deleteOnTerminationsArray,
+                Inputs.EbsInputs.BLOCK_DEVICE_MAPPING_DEVICE_NAMES_STRING, Inputs.EbsInputs.DELETE_ON_TERMINATIONS_STRING);
+        InputsUtil.validateAgainstDifferentArraysLength(deviceNamesArray, volumeIdsArray,
+                Inputs.EbsInputs.BLOCK_DEVICE_MAPPING_DEVICE_NAMES_STRING, Inputs.EbsInputs.VOLUME_IDS_STRING);
+        InputsUtil.validateAgainstDifferentArraysLength(deviceNamesArray, noDevicesArray,
+                Inputs.EbsInputs.BLOCK_DEVICE_MAPPING_DEVICE_NAMES_STRING, Inputs.EbsInputs.NO_DEVICES_STRING);
+        InputsUtil.validateAgainstDifferentArraysLength(deviceNamesArray, virtualNamesArray,
+                Inputs.EbsInputs.BLOCK_DEVICE_MAPPING_DEVICE_NAMES_STRING, Inputs.EbsInputs.BLOCK_DEVICE_MAPPING_VIRTUAL_NAMES_STRING);
+
```
Comment (gituser173) [inline cs-amazon/src/main/java/io/cloudslang/content/amazon/factory/helpers/InstanceUtils.java]: extract the code below as a new method named "setOptional" or something similar

File: cs-amazon/src/main/java/io/cloudslang/content/amazon/factory/helpers/InstanceUtils.java (hunk 1 above)
Comment (mihaitusa) [inline cs-amazon/src/main/java/io/cloudslang/content/amazon/factory/helpers/InstanceUtils.java]: ok

**Comment by gituser173** [comment]
looks good 👍

File: cs-amazon/src/main/java/io/cloudslang/content/amazon/factory/helpers/InstanceUtils.java (hunk 2 above)
Comment (mihaitusa) [inline cs-amazon/src/main/java/io/cloudslang/content/amazon/factory/helpers/InstanceUtils.java]: Done.

**Comment by mihaitusa** [comment]
@gituser173  If I add junits (trivial) can I merge it after?

**Comment by gituser173** [comment]
Yes

**Pull request merged**